gradio/pipelines.py

```diff
@@ -3,268 +3,77 @@
-                components.Image(type="filepath", label="Input Image", render=False),
-                components.Textbox(label="Question", render=False),
-            ],
-            "outputs": components.Label(label="Score", render=False),
-            "preprocess": lambda img, q: {"image": img, "question": q},
-            "postprocess": lambda r: {i["answer"]: i["score"] for i in r},
-        }
-    elif hasattr(transformers, "ImageToTextPipeline") and isinstance(
-        pipeline,
-        pipelines.image_to_text.ImageToTextPipeline,  # type: ignore
-    ):
-        pipeline_info = {
-            "inputs": components.Image(
-                type="filepath", label="Input Image", render=False
-            ),
-            "outputs": components.Textbox(label="Text", render=False),
-            "preprocess": lambda i: {"images": i},
-            "postprocess": lambda r: r[0]["generated_text"],
-        }
-    elif hasattr(transformers, "ObjectDetectionPipeline") and isinstance(
-        pipeline, pipelines.object_detection.ObjectDetectionPipeline
-    ):
-        pipeline_info = {
-            "inputs": components.Image(
-                type="filepath", label="Input Image", render=False
-            ),
-            "outputs": components.AnnotatedImage(
-                label="Objects Detected", render=False
-            ),
-            "preprocess": lambda i: {"inputs": i},
-            "postprocess": lambda r, img: (
-                img,
-                [
-                    (
-                        (
-                            i["box"]["xmin"],
-                            i["box"]["ymin"],
-                            i["box"]["xmax"],
-                            i["box"]["ymax"],
-                        ),
-                        i["label"],
-                    )
-                    for i in r
-                ],
-            ),
-        }
+    if str(type(pipeline)).startswith("<class 'transformers.pipelines"):
+        pipeline_info = _handle_transformers_pipeline(pipeline)
+
+    elif str(type(pipeline)).startswith("<class 'diffusers.pipelines"):
+        pipeline_info = _handle_diffusers_pipeline(pipeline)
+
     else:
-        raise ValueError(f"Unsupported pipeline type: {type(pipeline)}")
+        raise ValueError(
+            "pipeline must be a transformers.pipeline or diffusers.pipeline"
+        )
 
     # define the function that will be called by the Interface
     def fn(*params):
         data = pipeline_info["preprocess"](*params)
-        # special cases that needs to be handled differently
-        if isinstance(
-            pipeline,
-            (
-                pipelines.text_classification.TextClassificationPipeline,
-                pipelines.text2text_generation.Text2TextGenerationPipeline,
-                pipelines.text2text_generation.TranslationPipeline,
-            ),
-        ):
-            data = pipeline(*data)
-        else:
+        if "tranformers.pipelines" in str(type(pipeline)):
```

Comment:
I doubt this code path was ever exercised since there's a typo here.

Reply:
fixed this, will add tests.

Reply:
Fixed the typo and tried a transformer pipeline and diffuser pipeline locally and they are working fine.